Sources/Tracing/Docs.docc/Guides/InstrumentYourLibrary.md

```diff
@@ -0,0 +1,331 @@
+# Instrument Your Library or Framework
+
+## Overview 
+
+This guide is aimed at library and framework developers who wish to instrument their code using distributed tracing.
+
+Doing so within a library may enable automatic trace propagation and is key to propagating trace information across
+distributed nodes, e.g. by instrumenting the HTTP client used by such system.
+
+Other examples of libraries which would benefit _the most_ from being instrumented using distributed tracing include:
+
+- HTTP Clients (e.g. AsyncHTTPClient),
+- HTTP Servers (e.g. Vapor or Smoke),
+- RPC systems (Swift gRPC or Swift's `DistributedActorSystem` implementations),
+- database drivers (e.g. SQL or MongoDB clients),
+- any other library which can emit meaningful span information about tasks it is performing.
+
+The most important libraries to instrument are "edge" libraries, which serve to connect between systems, because
+it is them who must inject and extract contextual baggage metadata to enable distributed trace ``Span`` propagation.
+
+Following those, any database or other complex library which may be able to emit useful information about its internals are
+also good candidates to being instrumented. Note that libraries may do so optionally, or hide the "verboseness" of such traces
+behind options, or only attach information if a ``Span`` is already active etc. Please review your library's documentation to learn
+more about it has integrated tracing support.
+
+### Propagating baggage metadata
+
+When crossing boundaries between processes, such as making or receiving an HTTP request, the library responsible for doing so should invoke instrumentation in order to inject or extract the contextual baggage metadata into/from the "carrier" type (such as the `HTTPResponse`) type.
+
+#### Handling outbound requests
+
+When a library makes an "outgoing" request or message interaction, it should invoke the method of a configured instrument. This will invoke whichever instrument the end-user has configured and allow them to customize what metadata gets to be propagated. This can be depicted by the following diagram:
```

Comment:
```suggestion
When a library makes an "outgoing" request or message interaction, it should invoke the method of a configured instrument. This will invoke whichever instrument the end-user has configured and allow them to customize what metadata gets to be propagated. This is depicted in the following diagram:
```